### PR TITLE
refactor: user can execute show grants query itself

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

show grants will check the PrincipalIdentity. If user is current user or role is in curent user's roles, will skip privilege check.

- Fixes #[Link the issue here]

## Tests

- [x] Logic Test

## Type of change

- [x] Refactoring

### CODE REVIEW

**Review by flaneur2020 — APPROVED** [review]

File: src/query/service/src/interpreters/access/privilege_access.rs [hunk 1]
```diff
@@ -952,6 +952,30 @@ impl AccessChecker for PrivilegeAccess {
                 self.validate_access(&GrantObject::Global, UserPrivilegeType::Super)
                     .await?;
             }
+            Plan::ShowGrants(plan) => {
+                let current_user = self.ctx.get_current_user()?;
+                if let Some(principal) = &plan.principal {
+                   match principal {
+                       PrincipalIdentity::User(user) => {if current_user.identity() == *user {
```
Comment (BohuTANG) [inline src/query/service/src/interpreters/access/privilege_access.rs:959]: What's the code style here?

Comment (TCeason) [reply src/query/service/src/interpreters/access/privilege_access.rs:959]: Strange, I run cargo fmt on my local.

Comment (TCeason) [reply src/query/service/src/interpreters/access/privilege_access.rs]: Delete the {} in the newset commit.

Comment (BohuTANG) [reply src/query/service/src/interpreters/access/privilege_access.rs]: Not related to `{}`, my RustRover works fine:
image

**Pull request merged**